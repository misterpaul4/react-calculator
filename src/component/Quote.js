import React, { useState } from 'react';

const Quote = () => {
  const [quoteObj, generateQuote] = useState({
    text: 'The price of greatness is responsibility.',
    author: 'Winston Churchill',
  });

  const random = max => Math.floor(Math.random() * Math.floor(max));

  const handClick = () => {
    fetch('https://type.fit/api/quotes')
      .then(response => response.json())
      .then(data => {
        const rand = random(data.length);
        const quote = data[rand];
        generateQuote({
          author: quote.author,
          text: quote.text,
        });
      });
  };

  return (
    <div>
      <button type="button" onClick={handClick}>generate</button>
      <h1>Quote</h1>
      <p>{quoteObj.text}</p>
      <h1>{quoteObj.author ? 'Author' : ''}</h1>
      <p>{quoteObj.author}</p>
    </div>
  );
};

export default Quote;
