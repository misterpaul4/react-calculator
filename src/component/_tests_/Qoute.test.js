import RRender from 'react-test-renderer';
import Quote from '../Quote';

it('renders the Quote Page correctly without errors', () => {
  const tree = RRender.create(<Quote />).toJSON();

  expect(tree).toMatchSnapshot();
});
