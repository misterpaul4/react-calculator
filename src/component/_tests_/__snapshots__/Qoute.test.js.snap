// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the Quote Page 1`] = `
<p>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
</p>
`;

exports[`renders the Quote Page  1`] = `
<p>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
</p>
`;

exports[`renders the Quote Page 1`] = `
<p>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
</p>
`;

exports[`renders the Quote Page correctly without errors 1`] = `
<p>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
</p>
`;

exports[`renders the Quote Page wi 1`] = `
<p>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
</p>
`;
